vendor/vendor-ios/src/main/kotlin/com/malinskiy/marathon/ios/xctestrun/Xctestrun.kt

```diff
@@ -23,83 +23,123 @@ class Xctestrun(inputStream: InputStream) {
+
+    private fun testHostBundle(targetName: String): String? = targetKeyValue(targetName, XctestrunKey.TestBundlePath)
+
+    fun testHostBundlePath(targetName: String): String? {
+        return testHostBundle(targetName)?.let { testHostBundle ->
+            return testHostPath(targetName)?.let { testHostPath ->
+                // __TESTHOST__/PlugIns/sample-appUITests.xctest
+                // __TESTROOT__/Debug-iphonesimulator/sample-appUITests-Runner.app
+                return testHostBundle.replace("""^__TESTHOST__""".toRegex(),
+                        testHostPath.replace("""^__TESTROOT__/""".toRegex(), ""))
+            }
+        } ?: null
+    }
 
     /**
      * @see <a href="x-man-page://5/xcodebuild.xctestrun">xcodebuild.xctestrun(5)</a>
      */
-    val isUITestBundle = propertyList.valueForKeypath(target, PropertyListKey.IsUITestBundle) as Boolean
+    fun  isUITestBundle(targetName: String): Boolean? = targetKeyValue(targetName, XctestrunKey.IsUITestBundle)
 
     /**
      * @see <a href="x-man-page://5/xcodebuild.xctestrun">xcodebuild.xctestrun(5)</a>
      */
-    val environmentVariables = propertyList.valueForKeypath(target, PropertyListKey.EnvironmentVariables) as PropertyListMap
+    fun environmentVariables(targetName: String): PropertyListMap? = targetKeyValue(targetName, XctestrunKey.EnvironmentVariables)
 
     /**
      * @see <a href="x-man-page://5/xcodebuild.xctestrun">xcodebuild.xctestrun(5)</a>
      */
-    val testingEnvironmentVariables = propertyList.valueForKeypath(target, PropertyListKey.TestingEnvironmentVariables) as PropertyListMap
+    fun testingEnvironmentVariables(targetName: String): PropertyListMap? = targetKeyValue(targetName, XctestrunKey.TestingEnvironmentVariables)
 
     /**
      * @see <a href="x-man-page://5/xcodebuild.xctestrun">xcodebuild.xctestrun(5)</a>
      */
-    private val skipTestIdentifiers = propertyList.valueForKeypath(target, PropertyListKey.SkipTestIdentifiers) as Array<Any>
+    fun skipTestIdentifiers(targetName: String): Array<Any>? = targetKeyValue(targetName, XctestrunKey.SkipTestIdentifiers)
 
     /**
      * Returns `true` if specified test should be excluded from the test run.
      */
     @suppress("ReturnCount")
     fun isSkipped(test: Test): Boolean {
-        if (test.pkg != targetName) return false
+        val targetName = targetNameFromProductModuleName(test.pkg) ?: return false
+        val skipped = skipTestIdentifiers(targetName) ?: return false
 
-        val skippedMethods = skippedTestMethodsByClass[test.clazz] ?: return false
+        return skipped.contains(test.clazz) || skipped.contains("${test.clazz}/${test.method}")
+    }
 
-        return skippedMethods.isEmpty() || skippedMethods.contains(test.method)
+    private fun targetNameFromProductModuleName(productModuleName: String): String? {
+        return targetNames.firstOrNull {
+            //TODO: Double check
```

Comment:
Tests failed here because it throws exception since `first` is used. Changed to `firstOrNull`. Problem then is that the expected name is `sample-appUITests` but `productModuleName` returns `sample_appUITests`. We need to double check this